src/utils/DataConfig.py

```diff
@@ -0,0 +1,208 @@
+from utils.typing import assert_type
+from loader.load_opportunity_dataset import load_opportunity_dataset
+from loader.load_sonar_dataset import load_sonar_dataset
+import itertools
+import json
+import os
+
+import pandas as pd
+from dataclasses import dataclass
+
+
+@dataclass
+class DataConfig:
+    """
+    Config Interface:
+    (can be used generic)
+        data_config.activity_idx_to_activity_name(activity_idx) (subclasses need to define the mapping that is required for that)
+        data_config.load_dataset()
+    """
+
+    # Dataset Config (subclass responsibility) -----------
+
+    raw_label_to_activity_idx_map = None
+    activity_idx_to_activity_name_map = None
+
+
+    # interface (subclass responsibility to define) ------------------------------------------------------------
+
+    def load_dataset(self) -> "list[Recording]":
+        raise NotImplementedError(
+            "init subclass of Config that defines the method activity_idx_to_activity_name"
+        )
+
+    # generic
+    def raw_label_to_activity_idx(self, label: str) -> int:
+        """
+        from the label as it is saved in the dataset, to the activity index
+        (Relabeling)
+        """
+        assert (
+            self.raw_label_to_activity_idx_map is not None
+        ), "init subclass of Config that defines the var raw_label_to_activity_idx_map"
```

Comment:
[MINOR] Can we change this to
`A subclass of Config which initializes the var raw_label_to_activity_idx_map should be used to access activity mapping.`

I  think the current message is a bit confusing. What went wrong? The initialization of the subclass?